optional "failure_onion" in reply to htlc_accepted hook. (#4187)
ElementsProject/lightning: doc/PLUGINS.md

```

`fail` will tell `lightningd` to fail the HTLC with a given hex-encoded
`failure_message` (please refer to the [spec][bolt4-failure-messages] for
details: `incorrect_or_unknown_payment_details` is the most common).

```json
{
  "result": "fail",
  "failure_onion": "[292bytes of a serialized error packet]"
}
```

> If instead of `failure_message` the response contain a hex-encoded

Reviewer: ```suggestion
Instead of `failure_message` the response can contain a ...
```

`failure_onion` that will be used instead (please refer to the
[spec][bolt4-failure-onion] for details). Note that `lightningd` will
apply the obfuscation step to the value retrned here with its own shared
secret (and key type `ammag`) before returning it to the previous hop.

> 

Reviewer: Maybe you could also point out the 'LN bridging nodes' usecase for this here.

```

`fail` will tell `lightningd` to fail the HTLC with a given hex-encoded
`failure_message` (please refer to the [spec][bolt4-failure-messages] for
details: `incorrect_or_unknown_payment_details` is the most common).

```json
{
  "result": "fail",

>   "failure_onion": "[292bytes of a serialized error packet]"

Reviewer 2: The error onion has a variable size, so it'd be better not allude to a fixed size here by putting the `292bytes` here.

Author: Does it? I was pretty sure it had a fixed size.

This proves I shouldn't be writing these things.

Reviewer: You are for sure not more or less talented than I am :)
It's about learning not knowing

Author: Why has it a variable size? The spec says it is 260 bytes plus an HMAC of 32 bytes, then it is XORed against a cipherstream of the same size on every hop. I don't understand how it can be different than 292 bytes.

Author: Here: https://github.com/lightningnetwork/lightning-rfc/blob/master/04-onion-routing.md#returning-errors

It says the packet consists of `hmac(32) + failure_len(2) + failure + pad_len(2) + pad` such that `failure + pad = 256`. The `channel_update` and other things I imagine are contained in the `failure` element above.

}
```

Instead of `failure_message` the response can contain a hex-encoded
`failure_onion` that will be used instead (please refer to the
[spec][bolt4-failure-onion] for details). This can be used, for example,
if you're writing a bridge between two Lightning Networks. Note that

> `lightningd` will apply the obfuscation step to the value retrned here

Reviewer 2: ```suggestion
`lightningd` will apply the obfuscation step to the value returned here
```
